deltacat/compute/converter/pyiceberg/catalog.py

```diff
@@ -0,0 +1,73 @@
+from typing import Optional
+
+
+def load_catalog(iceberg_catalog_name, iceberg_catalog_properties):
+    catalog = load_catalog(
+        name=iceberg_catalog_name,
+        **iceberg_catalog_properties,
+    )
+    return catalog
+
+
+def get_s3_path(
+    bucket_name: str,
+    database_name: Optional[str] = None,
+    table_name: Optional[str] = None,
+) -> str:
+    result_path = f"s3://{bucket_name}"
+    if database_name is not None:
+        result_path += f"/{database_name}.db"
+
+    if table_name is not None:
+        result_path += f"/{table_name}"
+    return result_path
+
+
+def get_bucket_name():
+    return "metadata-py4j-zyiqin1"
+
+
+def get_s3_prefix():
+    return get_s3_path(get_bucket_name())
+
+
+def get_credential():
+    import boto3
+
+    boto3_session = boto3.Session()
+    credentials = boto3_session.get_credentials()
+    return credentials
+
+
+def get_glue_catalog():
+    #     # from pyiceberg.catalog.glue import GLUE_CATALOG_ENDPOINT, GlueCatalog
+    from pyiceberg.catalog import load_catalog
+
+    credential = get_credential()
+    access_key_id = credential.access_key
+    secret_access_key = credential.secret_key
+    session_token = credential.token
```

Comment:
get_credential() returns RefreshableCredentials. There is a race condition here where refresh happens when retrieving secret key which results in invalid access key, secret key pair. See more about this common problem here: https://stackoverflow.com/questions/36287720/boto3-get-credentials-dynamically